### PR TITLE
[cloud] ovirt: add headers param to auth

### PR DESCRIPTION
**Pull request by zuc**
##### SUMMARY
Adds a `headers` dict param to oVirt authentication.
This allows adding `Filter: true` http header to each API call (needed for non-admin users).

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ovirt module

##### ANSIBLE VERSION
```
ansible 2.3.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
```

##### ADDITIONAL INFORMATION
None

### CODE REVIEW

**Comment by machacekondra** [comment]
Which version of the oVirt do you use? Note that since 4.1.1[1], you can just do:

```
 # echo 'ENGINE_API_FILTER_BY_DEFAULT="true"' >  /etc/ovirt-engine/engine.conf.d/99-filter-by-default.conf 
 # systemctl restart ovirt-engine
```

With this enabled the normal user without admin permissions don't need to send `filter` header and will be work as the user sent `filter` header.

[1] https://gerrit.ovirt.org/#/c/74173/

**Comment by zuc** [comment]
@machacekondra We are running on a multi-tenant oVirt datacenter for which I don't have admin rights

File: lib/ansible/module_utils/ovirt.py [hunk 1]
```diff
@@ -143,6 +143,7 @@ def create_connection(auth):
         insecure=auth.get('insecure', False),
         token=auth.get('token', None),
         kerberos=auth.get('kerberos', None),
+        headers=dict(filter=auth.get('filter', False)),
```
Comment (machacekondra) [inline lib/ansible/module_utils/ovirt.py]: In order to work that properly in 4.2 where you don't have specify Filter header to user behave as user, we have to don't send anything if user didn't pass that. So I would suggest to have parameter `headers`, which is dictionary, rather then `Filter`.

Comment (zuc) [reply lib/ansible/module_utils/ovirt.py]: Good point, I just thought `filter` was more understandable.
I'll push a fix ASAP.

Comment (zuc) [reply lib/ansible/module_utils/ovirt.py]: @machacekondra Done!

**Review by machacekondra — APPROVED** [review]
Thanks a lot!

**Comment by machacekondra** [comment]
@mwperina @ryansb shipit

**Pull request merged**